### PR TITLE
add set interface, first map set implementation

### PR DESCRIPTION
**Pull request by nicbaz**

### CODE REVIEW

**Comment by elwinar** [comment]
Any reason not to use https://godoc.org/github.com/deckarep/golang-set ?

**Comment by elwinar** [comment]
/github.com/nicbaz/go-set/map_set.go
        Line 8: comment on exported type MapSet should be of the form "MapSet ..." (with optional leading article)
        Line 22: exported method MapSet.ForEach should have comment or be unexported
        Line 68: comment on exported method MapSet.Size should be of the form "Size ..."

(golint)

**Comment by nicbaz** [comment]
yep, really not a fan of those implementations. Whether it's the thread safety or the use of `map[type]bool`

**Comment by elwinar** [comment]
Generally speaking, I''m not fan of set implementations: they don't add any meaningful feature to what a map already provide, except for very specialized methods that a more often than not relevant only to the program that use them.

**Comment by nicbaz** [comment]
I agree to an extent : map are a pain to manipulate in that context especially when you have to add/remove large sets of values. I'd rather have some sugar around, plus this is a naive implementation and I'm gonna need faster and ordered implementations soon.

**Comment by nicbaz** [comment]
fixed the golint issue, added travis-ci with `golint`, `vet` and `gofmt`.

**Comment by nicbaz** [comment]
will add coveralls but that's for another commit.

File: map_set.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	return
+}
+
+// IsEmpty returns true if the set contains no value, false otherwise.
+func (set MapSet) IsEmpty() bool {
+	return 0 == len(set)
+}
+
+// Join concatenates all values of the set using separator between them in the resulting string.
+func (set MapSet) Join(separator string) string {
```
Comment (elwinar) [inline map_set.go]: We should use an interface to allow the user to control how each item is rendered.
Something like that:

```
if value, ok := value.(fmt.Stringer) {
    buffer.WriteString(value.String())
    buffer.WriteString(separator)
}
```

**Comment by elwinar** [comment]
We should use an interface to allow the user to control how each item is rendered.
Something like that:

```
if value, ok := value.(fmt.Stringer) {
    buffer.WriteString(value.String())
    buffer.WriteString(separator)
}
```

**Comment by nicbaz** [comment]
good idea, next iteration.

**Pull request merged**